src/main/java/jenkins/plugins/logstash/configuration/Syslog.java

```diff
@@ -0,0 +1,75 @@
+package jenkins.plugins.logstash.configuration;
+
+import org.kohsuke.stapler.DataBoundConstructor;
+import org.kohsuke.stapler.DataBoundSetter;
+
+import com.cloudbees.syslog.MessageFormat;
+
+import hudson.Extension;
+import jenkins.plugins.logstash.persistence.LogstashIndexerDao.SyslogProtocol;
+import jenkins.plugins.logstash.persistence.SyslogDao;
+
+public class Syslog extends LogstashIndexer<SyslogDao>
+{
+  private MessageFormat messageFormat;
+  private SyslogProtocol syslogProtocol;
+
+  @DataBoundConstructor
+  public Syslog()
+  {
+  }
+
+  public MessageFormat getMessageFormat()
+  {
+    return messageFormat;
+  }
+
+  @DataBoundSetter
+  public void setMessageFormat(MessageFormat messageFormat)
+  {
+    this.messageFormat = messageFormat;
+  }
+
+  public SyslogProtocol getSyslogProtocol()
+  {
+    return syslogProtocol;
+  }
+
+  @DataBoundSetter()
+  public void setSyslogProtocol(SyslogProtocol syslogProtocol)
+  {
+    this.syslogProtocol = syslogProtocol;
+  }
+
+  @Override
+  protected boolean shouldRefreshInstance()
+  {
+    return super.shouldRefreshInstance() ||
+        !instance.getMessageFormat().equals(messageFormat);
```

Comment:
we are not checking the protocol -- is this intentional?

Reply:
The protocol is currently not used. Maybe we can remove this completely. Or is there an other protocol than UDP possible.

Reply:
There were attempts to implement TCP in the past: #26

I'd either remove the field or use it in comparison to avoid future confusion.

Reply:
will add it to comparison. We should also implement the tcp protocol I think, just for completeness